hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/OverwriteWithLatestHiveRecordMerger.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.hadoop;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+
+import org.apache.avro.Schema;
+
+import java.io.IOException;
+
+public class OverwriteWithLatestHiveRecordMerger extends HoodieHiveRecordMerger {
```

Comment:
Should the classes be reorganized as:
- `HoodieHiveRecordMerger` which is an abstract class implementing `getRecordType()`
- `DefaultHiveRecordMerger` which implements merger strategy `DEFAULT_MERGER_STRATEGY_UUID`
- `OverwriteWithLatestHiveRecordMerger` which implements `OVERWRITE_MERGER_STRATEGY_UUID`

Reply:
it's organized the same as spark. IDK if we can change the spark organization without messing people up. And idk if better organization in hive is worth having it be different than spark

Reply:
Given we're introducing the new record merger classes in the next release, it's OK to re-organize it for better understanding.  For Spark, I think it's OK to change the class structure and possibly rename the classes, as we don't expect users to explicitly set the merger class, unless it's custom.